java/java.editor/test/unit/src/org/netbeans/modules/editor/java/GoToSupportTest.java

```diff
@@ -1581,7 +1581,10 @@ private static boolean hasRecords() {
     }
 
     public void testJavadocSnippetHighlightRecord() throws Exception {
-        if (!TreeShims.isJDKVersionRelease18_Or_Above()) {
+        try {
+            SourceVersion.valueOf("RELEASE_18"); //NOI18N
+        } catch (IllegalArgumentException ex) {
+            //OK, no RELEASE_18, skip tests
             return;
         }
```

Comment:
does this mean the tests don't use nb-javac? Would it be better to enable nb-javac?

Reply:
If nb-javac-18 plugin installed & enabled it uses nb-javac otherwise it uses the JDK Source version. If JDK source version is below 18 then this test won't be execute.

Reply:
yes. but tests can be configured to use nb-javac.
looking at the properties:
https://github.com/apache/netbeans/blob/master/java/java.editor/nbproject/project.properties#L113-L114

it looks like nb-javac is already enabled for the tests. It should work without this check I believe.

Reply:
Yes, then I believe we can remove other conditional checks also of this class are _SourceVersion.valueOf("RELEASE_10");_ and _SourceVersion.valueOf("RELEASE_14");_

Reply:
yep, it should be possible to remove the javac version checks and the reflective code and change everything to direct calls.

Even if the user uninstalls nb-javac, NB has a runtime dependency check which will tell the user to run on a specific JDK or install nb-javac again. You can see it here: https://github.com/apache/netbeans/pull/3953/files